src/hera/workflows/cluster_workflow_template.py

```diff
@@ -0,0 +1,70 @@
+from hera.exceptions import NotFound
+from hera.workflows.models import (
+    ClusterWorkflowTemplate as _ModelClusterWorkflowTemplate,
+    ClusterWorkflowTemplateCreateRequest,
+    ClusterWorkflowTemplateLintRequest,
+    ClusterWorkflowTemplateUpdateRequest,
+)
+from hera.workflows.protocol import TWorkflow
+from hera.workflows.workflow_template import WorkflowTemplate
+
+
+class ClusterWorkflowTemplate(WorkflowTemplate):
+    """ClusterWorkflowTemplates are WorkflowTemplates that are cluster scoped, meaning they are accessible from all
+    namespaces.
+    """
+
+    def create(self) -> TWorkflow:  # type: ignore
+        """Creates the ClusterWorkflowTemplate on the Argo cluster."""
+        assert self.workflows_service, "workflow service not initialized"
+        assert self.namespace, "workflow namespace not defined"
+        return self.workflows_service.create_cluster_workflow_template(
+            ClusterWorkflowTemplateCreateRequest(template=self.build())
+        )
+
+    def get(self) -> TWorkflow:
```

Comment:
Should `get`s on classes such as workflows, workflow template, etc, be class methods? 🤔 Or static methods that return a new `Workflow` based on the details of the upstream object if one is found? The service _does_ have to be instantiated, so I understand if this implementation is a _necessary_ approach

Reply:
Hmm yeah maybe this should be a private method (as it's needed for `update`), and a `get` could be a static method to get via a `name` (+any other info needed)

Reply:
Well, we have `WorkflowsService.get_workflow` with more explicit parameters? Maybe we just keep this `Workflow.get` private

Reply:
We already released the `WorkflowTemplate.get` method in v5.3.0, from #642 [here](https://github.com/argoproj-labs/hera/pull/642/files#diff-578cbe46ffddea5458bdb9b1701e5ec32fe9ef4b8bb5334c5c09791e77da7ce0R42) - I need to use the same interface to override it and avoid calling get_workflow_template instead of \*\_cluster_*. Tbh I don't think it's a "mistake" to add it to the public interface, but now that we have we should probably leave it as is.

Reply:
I think we could introduce a kwarg called `cluster: bool = False` to support creating cluster level workflow templates while maintaining backwards compatibility 🤔 thoughts?